python/analysis/vector/qgsgeometrysnapper.sip

```diff
@@ -0,0 +1,43 @@
+/**
+ * \class QgsGeometrySnapper
+ * \ingroup analysis
+ * QgsGeometrySnapper allows a geometry to be snapped to the geometries within a
+ * different refence layer. Vertices in the geometries will be modified to
+ * match the reference layer features within a specified snap tolerance.
+ * \note added in QGIS 3.0
+ */
+
+class QgsGeometrySnapper : QObject
+{
+%TypeHeaderCode
+#include <qgsgeometrysnapper.h>
+%End
+
+  public:
+
+    /**
+     * Constructor for QgsGeometrySnapper. A reference layer which contains geometries to snap to must be
+     * set. The snap tolerance is specified in the layer units for the
+     * reference layer, and it is assumed that all geometries snapped using this object will have the
+     * same CRS as the reference layer (ie, no reprojection is performed).
+     */
+    QgsGeometrySnapper( QgsVectorLayer* referenceLayer, double snapTolerance );
+
+    /**
+     * Snaps a geometry to the reference layer and returns the result. The geometry must be in the same
+     * CRS as the reference layer.
+     */
+    QgsGeometry snapGeometry( const QgsGeometry& geometry ) const;
+
+    /**
+     * Snaps a set of features to the reference layer and returns the result. This operation is
+     * multithreaded for performance. The featureSnapped() signal will be emitted each time a feature
+     * is processed. This method is not safe to call from multiple threads concurrently.
```

Comment:
I wonder if we should stick to the Qt terminology for thread-safety:

http://doc.qt.io/qt-4.8/threads-reentrancy.html

> Throughout the documentation, the terms reentrant and thread-safe are used to mark classes and functions to indicate how they can be used in multithread applications:
* A thread-safe function can be called simultaneously from multiple threads, even when the invocations use shared data, because all references to the shared data are serialized.
* A reentrant function can also be called simultaneously from multiple threads, but only if each invocation uses its own data.